website/generate.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Builds the ParlAI website.
+"""
+
+import os
+import git
+import markdown
+from mdx_gfm import PartialGithubFlavoredMarkdownExtension
+
+GIT_ROOT_LEVEL = git.Git().rev_parse('--show-toplevel')
+WEBSITE_ROOT = os.path.join(GIT_ROOT_LEVEL, 'website')
+TEMPLATES = os.path.join(WEBSITE_ROOT, 'templates')
+OUT_DIR = os.path.join(WEBSITE_ROOT, 'build')
+
+
+def ghmarkdown(source):
+    return markdown.markdown(
+        source,
+        extensions=[PartialGithubFlavoredMarkdownExtension()]
+    )
+
+
+def _read_file(filename):
+    with open(filename) as f:
+        return f.read()
+
+
+def _mkdirp(directory):
+    """Equivalent to mkdir -p"""
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def _write_file(partial_filename, content):
+    filename = os.path.join(OUT_DIR, partial_filename)
+
+    _mkdirp(os.path.dirname(filename))
+
+    print("writing {} bytes to {}".format(len(content), filename))
+    with open(filename, 'w') as f:
+        f.write(content)
+
+
+def wrap_base(content, title):
+    template = _read_file(os.path.join(TEMPLATES, 'base.html'))
+    template = template.replace('{{{CONTENT}}}', content)
+    template = template.replace('{{{TITLE}}}', title)
+    return template
+
+
+def make_errorpage():
+    content = _read_file(os.path.join(TEMPLATES, 'error.html'))
+    html = wrap_base(content, "Error")
+    _write_file('error.html', html)
+
+
+def make_aboutpage():
+    template = _read_file(os.path.join(TEMPLATES, 'about.html'))
+    readme = _read_file(os.path.join(GIT_ROOT_LEVEL, 'README.md'))
+    readme_html = ghmarkdown(readme)
+    readme_html = readme_html.replace("docs/source/\\", "/docs/")
```

Comment:
I also see some links in the page for `/static/docs/` e.g. I see links for these tutorials `http://parl.ai/static/docs/tutorial_mturk.html` and `http://parl.ai/static/docs/tutorial_messenger.html` currently on the page.

Will these break with this new site? Is it also necessary to replace `/static/docs` with `/docs`?

Reply:
Oh man they totally will. I'll keep the old URL then.

Reply:
Resolving.

For posterity: we set up an S3 redirect.